### PR TITLE
[Numpy] Fix Tokenizers with the latest HuggingFace Tokenizer + Add scheduled test + Fix data cli

### PR DESCRIPTION
**Pull request by sxjscience**
- [x] `nlp_preprocess` --> `nlp_process`
- [x] Split different tokenizers to different files and support tokenizers==0.8.1 (Suggested reviewer @hymzoque )
- [x] Fix the lazy import strategy of data cli (Suggested reviewer @ZheyuYe)
- [x] Fix submit-job.py (Suggested reviewer @leezu or @barry-jin or @szha )
- [x] Add unittests for data CLI: learn_subword and apply_subword. The encoded results of different tokenizers will be uploaded as artifacts.

### CODE REVIEW

**Review by ZheyuYe — COMMENTED** [review]
The lazy import part looks good for me

**Review by hymzoque — APPROVED** [review]

File: tools/batch/docker/Dockerfile.cpu [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  rm -rf /var/lib/apt/lists/*
+
+RUN pip3 install --upgrade pip
+RUN pip3 install  --no-cache --upgrade \
+    wheel \
+    cmake \
+    awscli
+RUN git clone https://github.com/dmlc/gluon-nlp
+WORKDIR gluon-nlp
+ADD gluon_nlp_job.sh .
```
Comment (sxjscience) [inline tools/batch/docker/Dockerfile.cpu:32]: @szha I will first try to use `Dockerfile.cpu` and `Dockerfile.gpu`.

File: tools/batch/docker/Dockerfile.cpu [hunk 2]
```diff
@@ -0,0 +1,32 @@
+FROM python:3
```
Comment (szha) [inline tools/batch/docker/Dockerfile.cpu:1]:
```suggestion
FROM ubuntu:18.04
```

File: tools/batch/docker/README.md [hunk 3]
```diff
@@ -12,11 +12,14 @@ update the docker:
...
 
 # builds the Dockerfile as gluon-nlp-1 docker.
-docker build -t gluon-nlp-1 .
+docker build -f Dockerfile.gpu -t gluon-nlp-1:gpu .
+docker build -f Dockerfile.cpu -t gluon-nlp-1:cpu .
 
 # tags the recent build as gluon-nlp-1:latest, which AWS batch pulls from.
-docker tag gluon-nlp-1:latest 747303060528.dkr.ecr.us-east-1.amazonaws.com/gluon-nlp-1:latest
+docker tag gluon-nlp-1:gpu 747303060528.dkr.ecr.us-east-1.amazonaws.com/gluon-nlp-1:latest
+docker tag gluon-nlp-1:cpu 747303060528.dkr.ecr.us-east-1.amazonaws.com/gluon-nlp-1:cpu-latest
```
Comment (szha) [inline tools/batch/docker/README.md:20]: tagging won't work as is

File: tools/batch/docker/gluon_nlp_cpu_job.sh [hunk 4]
```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
```
Comment (szha) [inline tools/batch/docker/gluon_nlp_cpu_job.sh:1]: this file should be marked executable through +x

**Pull request merged**